scripts/mh_windowfill.pl

```diff
@@ -0,0 +1,244 @@
+	'description' => 'fill windows so scrolling starts bottom-up instead of top-down',
+	'license'     => 'BSD',
+	'authors'     => 'Michael Hansen',
+	'contact'     => 'mh on IRCnet #help',
+	'url'         => 'irc://open.ircnet.net',
+);
+
+##############################################################################
+#
+# global variables
+#
+##############################################################################
+
+our $windowfill_running = 0;
+
+##############################################################################
+#
+# common support functions
+#
+##############################################################################
+
+sub trim_spacelike($)
+{
+   my ($string) = @_;
+
+   if (defined($string))
+   {
+      $string =~ s/^\s+//g;
+      $string =~ s/\s+$//g;
+
+   } else {
+
+      $string = '';
+   }
+
+   return($string);
+}
+
+##############################################################################
+#
+# script functions
+#
+##############################################################################
+
+sub windowfill($)
+{
+	my ($window) = @_;
+
+	if (ref($window) ne 'Irssi::UI::Window')
+	{
+		die();
+	}
+
+	#
+	# fill window with empty lines and move already printed lines to the bottom
+	#
+	if (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+	{
+
+		while (($window->view()->{'ypos'} + 2) <= $window->{'height'})
+		{
+			$window->print('', MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+		}
+
+		my $linecount  = $window->{'height'};
+		my $line       = $window->view()->get_lines();
+
+		while ((ref($line) eq 'Irssi::TextUI::Line') and $linecount)
+		{
+			my $linetext = $line->get_text(1);
+
+			if ($linetext ne '')
+			{
+				# reprint line
+				$window->print($linetext, MSGLEVEL_CLIENTCRAP | MSGLEVEL_NEVER | MSGLEVEL_NO_ACT | MSGLEVEL_NOHILIGHT);
+				$line = $line->next();
+				$window->view()->remove_line($line->prev());
+
+			} else {
+
+				# skip empty line
+				$line = $line->next();
+			}
+
+			$linecount--;
+		}
+	}
+
+	return(1);
+}
+
+sub windowfill_all()
+{
+	#
+	# fill all windows with empty lines
+	#
+	for my $window (Irssi::windows())
+	{
+		if (ref($window) ne 'Irssi::UI::Window')
+		{
+			die();
+		}
+
+		windowfill($window);
+	}
+
+	return(1);
+}
+
+##############################################################################
+#
+# irssi signal handlers
+#
+##############################################################################
+
+sub signal_mainwindow_resized_last()
+{
+	if ($windowfill_running)
+	{
+		#
+		# fill all windows with empty lines
+		#
+		windowfill_all();
+	}
+
+	return(1);
+}
+
+Irssi::signal_add_last('mainwindow resized', 'signal_mainwindow_resized_last');
+
+sub signal_window_created_last($)
+{
+	my ($window) = @_;
+
+	if (ref($window) ne 'Irssi::UI::Window')
+	{
+		die();
+	}
+
+	if ($windowfill_running)
+	{
+		#
+		# fill created window with empty lines
+		#
+		windowfill($window);
+	}
+
+	return(1);
```

Comment:
What if i decide to call the function at a later date and check the return value?